cdk/l2-cdk-resources/third-party-integration/src/util.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ThirdPartyIntegrationProps } from './thirdPartyIntegrationBase';
+
+/**
+ * This method validates that all required properties are present.
+ * @param props
+ */
+export const validate = (props: ThirdPartyIntegrationProps) => {
+  if (!props.projectId) {
+    throw Error(getPropUndefinedMsg('projectId'));
+  }
+
+  if (!props.apiKeys.publicKey) {
+    throw Error(getPropUndefinedMsg('publicKey'));
+  }
+
+  if (!props.apiKeys.privateKey) {
+    throw Error(getPropUndefinedMsg('privateKey'));
+  }
```

Comment:
[nit] Nice to put all validation for an object (in this case `props.apiKeys`) in its own helper method, so it can be reused & modified in isolation.

Reply:
added separate helper method.